libraries/Microsoft.Bot.Builder.Ai.LUIS/LuisRecognizer.cs

```diff
@@ -10,53 +10,40 @@
 using Microsoft.Cognitive.LUIS.Models;
 using Newtonsoft.Json.Linq;
 using System.Text.RegularExpressions;
+using Microsoft.Bot.Builder.Core.Extensions;
 
 namespace Microsoft.Bot.Builder.Ai.LUIS
 {
     /// <inheritdoc />
     /// <summary>
-    /// Provides a LUIS-based implementation for the <see cref="IRecognizer"/> interface.
+    /// A LUIS based implementation of IRecognizer.
     /// </summary>
-    public class LuisRecognizer : ILuisRecognizer
+    public class LuisRecognizer : IRecognizer
     {
         private readonly LuisService _luisService;
         private readonly ILuisOptions _luisOptions;
         private readonly ILuisRecognizerOptions _luisRecognizerOptions;
         private const string MetadataKey = "$instance";
 
-        /// <summary>
-        /// Creates a new <see cref="LuisRecognizer"/> object.
-        /// </summary>
-        /// <param name="luisModel">The LUIS model to use to recognize text.</param>
-        /// <param name="luisRecognizerOptions">The LUIS recognizer options to use.</param>
-        /// <param name="options">The LUIS request options to use.</param>
         public LuisRecognizer(ILuisModel luisModel, ILuisRecognizerOptions luisRecognizerOptions = null, ILuisOptions options = null)
         {
             _luisService = new LuisService(luisModel);
             _luisOptions = options ?? new LuisRequest();
             _luisRecognizerOptions = luisRecognizerOptions ?? new LuisRecognizerOptions { Verbose = true };
         }
 
-        /// <summary>
-        /// Runs an utterance through a LUIS recognizer and returns the recognizer results.
-        /// </summary>
-        /// <param name="utterance">The utterance.</param>
-        /// <param name="ct">A cancellation token.</param>
-        /// <returns>The recognizer results.</returns>
+        /// <inheritdoc />
         public async Task<RecognizerResult> Recognize(string utterance, CancellationToken ct)
         {
-            var result = await CallAndRecognize(utterance, ct).ConfigureAwait(false);
-            return result.recognizerResult;
+            return (await CallAndRecognize(utterance, ct).ConfigureAwait(false));
         }
 
-        /// <summary>
-        /// Runs an utterance through a LUIS recognizer and returns the recognizer results.
-        /// </summary>
-        /// <param name="utterance">The utterance.</param>
-        /// <param name="ct">A cancellation token.</param>
-        /// <returns>The recognizer results and LUIS result.</returns>
-        /// <remarks>This method adds metadata to the recognizer's results.</remarks>
-        public Task<(RecognizerResult recognizerResult, LuisResult luisResult)> CallAndRecognize(string utterance, CancellationToken ct)
+        public async Task<T> Recognize<T>(string utterance, CancellationToken ct)
```

Comment:
Are value types legit here? Can I call:
int x = Recognize<int>("fifty four", null) ? 

I don't think that's legit, and a constraint on T would help that.